modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/model/listener/UserModelListener.java

```diff
@@ -0,0 +1,130 @@
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.portal.workflow.metrics.internal.model.listener;
+
+import com.liferay.portal.kernel.exception.ModelListenerException;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.ModelListener;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.search.engine.adapter.SearchEngineAdapter;
+import com.liferay.portal.search.engine.adapter.document.UpdateByQueryDocumentRequest;
+import com.liferay.portal.search.query.BooleanQuery;
+import com.liferay.portal.search.query.Queries;
+import com.liferay.portal.search.query.TermsQuery;
+import com.liferay.portal.search.script.ScriptBuilder;
+import com.liferay.portal.search.script.ScriptType;
+import com.liferay.portal.search.script.Scripts;
+import com.liferay.portal.workflow.metrics.internal.petra.executor.WorkflowMetricsPortalExecutor;
+import com.liferay.portal.workflow.metrics.internal.search.index.WorkflowMetricsIndex;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+/**
+ * @author Feliphe Marinho
+ */
+@Component(immediate = true, service = ModelListener.class)
+public class UserModelListener extends BaseModelListener<User> {
+
+	@Override
+	public void onAfterUpdate(User user) throws ModelListenerException {
+		if (_nameModified) {
+			ScriptBuilder builder = _scripts.builder();
+
+			builder.idOrCode(
+				StringUtil.read(
+					getClass(),
+					"dependencies/workflow-metrics-update-task-assignee-" +
+						"script.painless")
+			).language(
+				"painless"
+			).putParameter(
+				"userId", user.getUserId()
+			).putParameter(
+				"userName", user.getFullName()
+			).scriptType(
+				ScriptType.INLINE
+			);
+
+			BooleanQuery nestedBooleanQuery = _queries.booleanQuery();
+
+			TermsQuery termsQuery = _queries.terms("tasks.assigneeIds");
+
+			termsQuery.addValues(
+				Stream.of(
+					user.getUserId()
+				).map(
+					String::valueOf
+				).toArray(
+					Object[]::new
+				));
+
+			nestedBooleanQuery.addMustQueryClauses(termsQuery);
+
+			nestedBooleanQuery.addMustQueryClauses(
+				_queries.term("tasks.assigneeType", User.class.getName()));
+
+			_workflowMetricsPortalExecutor.execute(
+				() -> searchEngineAdapter.execute(
```

Comment:
@MarinhoFeliphe move the whole logic to create the UpdateByQueryDocumentRequest to the `Callable`.